database/src/main/scala/cromwell/database/slick/CallCachingSlickDatabase.scala

```diff
@@ -37,20 +37,12 @@ trait CallCachingSlickDatabase extends CallCachingStore {
     }
   }
 
-  override def fetchCachedResult(metaInfoId: MetaInfoId)(implicit ec: ExecutionContext): Future[Option[CachedResult]] = {
+  override def fetchCachedResult(metaInfoId: MetaInfoId)(implicit ec: ExecutionContext): Future[CachedResult] = {
     val action = for {
-      metaInfo <- dataAccess.metaInfoById(metaInfoId.id).result
+      metaInfo <- dataAccess.metaInfoById(metaInfoId.id).result.head
```

Comment:
per in-person discussion, I'll drop a comment here with why we currently believe this to be an invariant